packages/bun-plugin-svelte/src/index.ts

```diff
@@ -0,0 +1,91 @@
+import type { BunPlugin, BuildConfig, OnLoadResult } from "bun";
+import { strict as assert } from "node:assert";
+import { basename } from "node:path";
+import { compile, compileModule } from "svelte/compiler";
+import { getBaseCompileOptions, validateOptions, type SvelteOptions, hash } from "./options";
+
+const kEmptyObject = Object.create(null);
+const virtualNamespace = "bun-svelte";
+
+function SveltePlugin(options: SvelteOptions = kEmptyObject as SvelteOptions): BunPlugin {
+  if (options != null) validateOptions(options);
+
+  /**
+   * import specifier -> CSS source code
+   */
+  const virtualCssModules = new Map<string, string>();
+
+  return {
+    name: "bun-plugin-svelte",
+    setup(builder) {
+      const { config = kEmptyObject as Partial<BuildConfig> } = builder;
+      const baseCompileOptions = getBaseCompileOptions(options ?? (kEmptyObject as Partial<SvelteOptions>), config);
+
+      builder
+        .onLoad({ filter: /\.svelte(?:\.[tj]s)?$/ }, async args => {
+          assert(args && typeof args === "object");
+          const { path } = args;
+          assert(typeof path === "string");
+
+          var isModule = false;
+
+          switch (path.substring(path.length - 2)) {
+            case "js":
+            case "ts":
+              isModule = true;
+              break;
+          }
+
+          const sourceText = await Bun.file(path).text();
+
+          const side =
+            args && "side" in args // "side" only passed when run from dev server
```

Comment:
we probably should make this always specified

Reply:
agreed; this is out of scope for this PR